pkg/internal/computeresources/transformer_test.go

```diff
@@ -946,13 +1052,13 @@ func TestTransformerMultipleContainer(t *testing.T) {
 				Name: "sidecar-fizz",
 				Resources: corev1.ResourceRequirements{
 					Limits: corev1.ResourceList{
-						corev1.ResourceCPU:              resource.MustParse("4"),
+						corev1.ResourceCPU:              resource.MustParse("2"),
```

Comment:
The LimitRange in this test case has a "default" of 2 but not a "max" of 2, so I think we want to leave the limit at 4.

Reply:
got this. so it means we actually want to overwrite the default limits whenever there are limits from containers right? 
I will update the setLimits as the "overwrite with default only if there are no limits on containers"

Reply:
yes, that's right. the [docs](https://github.com/tektoncd/pipeline/blob/main/docs/compute-resources.md#limits) describe our intended behavior

Reply:
checked the code - the "default" in LimitRange was used to compute the `defaultContainerLimits`
https://github.com/tektoncd/pipeline/blob/99b6692a5cca86926650387830d05551236c323d/pkg/internal/computeresources/transformer.go#L193

in this case, by the doc, we still should keep the min of limits right?
so here - the defaultLimits as "4" and sidecar limits as "2", the expected is "2"

Reply:
I think it's actually the other way around here. The default limits are 2 and the sidecar limits are 4.

Reply:
your are correct. the "4" is from the sidecar. 
so the result "2" comes from the lower end of both limits.

Reply:
We don't want to allow default limits to override limits set on the container. Only the limitrange max should override limits set on the container.

Reply:
right, think your understanding matches the doc we are referring. 
just the code causing me a bit confused: https://github.com/tektoncd/pipeline/blob/4c775883dcb54431574343ac85be7423711f3e72/pkg/internal/computeresources/transformer.go#L197

here the max of LimitRange is used to compute the default limit (if there is no default) - so if we only allow override limits with default, think will need to split into 2 funcs - getDefaultLimit and getMaxLimit from LimitRange - so clear the "default" here we used

Reply:
got it-- let's make sure we have the test cases how we want them, and then we can change the code any way that's needed so we get the right results. If you need to split into 2 functions that's OK; for the review right now I'm just focusing on getting the test cases right.

Reply:
understood your logic. thanks Lee!

Reply:
kept the default as it was (since the max will be considered as the default if only max)
add a maxLimit getter func - so only take the min when max exists